pkg/storage/stores/local/boltdb_index_client.go

```diff
@@ -0,0 +1,73 @@
+package local
+
+import (
+	"context"
+	"sync"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/local"
+	chunk_util "github.com/cortexproject/cortex/pkg/chunk/util"
+	"go.etcd.io/bbolt"
+)
+
+type BoltdbIndexClientWithArchiver struct {
+	*local.BoltIndexClient
+
+	archiver *Archiver
+
+	done chan struct{}
+	wait sync.WaitGroup
+}
+
+// NewBoltDBIndexClient creates a new IndexClient that used BoltDB.
+func NewBoltDBIndexClient(cfg local.BoltDBConfig, archiveStoreClient chunk.ObjectClient, archiverCfg ArchiverConfig) (chunk.IndexClient, error) {
+	boltDBIndexClient, err := local.NewBoltDBIndexClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	archiver, err := NewArchiver(archiverCfg, archiveStoreClient, func(name string, operation int) (db *bbolt.DB, e error) {
+		return boltDBIndexClient.GetDB(name, operation)
```

Comment:
This is a good candidate for an interface. I think it make more sense then a function.

Reply:
Sorry, I am not sure why we need an interface here? We would only ever have 1 implementation here and it looks ugly to pass `boltDBIndexClient` to `Archiver` which itself embeds the `Archiver`. Maybe I am wrong but it would be good to hear your thoughts.

Reply:
```
type BoltDBGetter interface {
 GetLocal(name string, operation int) (*bbolt.DB,error)
}
```

It will easier for you to work with. I don't have strong opinions on that one. I just like to use interface for decoupling. This is just a nit.